### PR TITLE
[ATO-1685] Add SlotSet step in story with active_loop:null mapping condition

### PR DESCRIPTION
**Pull request by Tawakalt**
**Proposed changes**:
- Fixes [ATO-1685](https://rasahq.atlassian.net/browse/ATO-1685)

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

[ATO-1685]: https://rasahq.atlassian.net/browse/ATO-1685?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ancalita — COMMENTED** [review]
LGTM 🙌🏻 Could you also please look into adding an integration test - e.g. running the validate stories functionality on a story that has an intent with entities + domain with the same `from_entity`  slot with active_loop as null in the mapping?

File: data/test/test_integration/domain_with_null_active_loop.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    type: categorical
+    influence_conversation: true
+    values:
+      - savings
+      - checking
+      - credit
+    mappings: 
+    - type: from_entity
+      entity: account_type
+    # uncomment below to make the story conflict happen
```
Comment (ancalita) [inline data/test/test_integration/domain_with_null_active_loop.yml:27]: This comment is confusing, why would a story conflict happen if you comment out conditions with the current bugfix?

File: tests/integration_tests/core/test_cli_response.py [hunk 2]
```diff
@@ -148,3 +148,45 @@ def test_rasa_validate_quiet_no_errors(
...
     "https://rasa.com/docs/rasa/telemetry/telemetry." in output_text
+
+
+def test_rasa_validate_null_active_loop_no_errors(
+    run: Callable[..., RunResult], request: FixtureRequest
+):
+    # Test captures the subprocess output for the command run
+    # and validates that the data in 'data/test/test_integration' throws no cli errors
+
+    test_data_dir = Path(request.config.rootdir, "data", "test", "test_integration")
```
Comment (ancalita) [inline tests/integration_tests/core/test_cli_response.py:159]: I don't see a stories file in this directory, is it missing?

File: tests/integration_tests/core/test_cli_response.py [hunk 3]
```diff
@@ -148,3 +148,45 @@ def test_rasa_validate_quiet_no_errors(
...
+    assert "No story structure conflicts found." in stderr_text
+
+    output_text = "".join(result.outlines)
+    assert "Rasa Open Source reports anonymous usage telemetry"
+    "to help improve the product" in output_text
+    assert "for all its users." in output_text
+    assert "If you'd like to opt-out,"
+    "you can use `rasa telemetry disable`." in output_text
+    assert "To learn more, check out"
+    "https://rasa.com/docs/rasa/telemetry/telemetry." in output_text
```
Comment (ancalita) [inline tests/integration_tests/core/test_cli_response.py:192]: These asserts are not required, you can remove them.

**Review by ancalita — COMMENTED** [review]
Requested a clarification on the domain file, I also think the story is missing from the data subdir.

File: data/test/test_integration/domain_with_null_active_loop.yml (hunk 1 above)
Comment (Tawakalt) [reply data/test/test_integration/domain_with_null_active_loop.yml:27]: I forgot to remove that. I copied the domain content from the reproduce-bot.

File: tests/integration_tests/core/test_cli_response.py (hunk 2 above)
Comment (Tawakalt) [reply tests/integration_tests/core/test_cli_response.py:159]: It's in the sub data directory. Added in line 161.

**Review by ancalita — APPROVED** [review]
Nice work 💯 Thanks for adding the integration test 🎉

**Comment by Tawakalt** [comment]
> Nice work 💯 Thanks for adding the integration test 🎉

Thanks for the speedy review 🚀

**Pull request merged**